pkg/operator/ceph/disruption/machinedisruption/reconcile.go

```diff
@@ -0,0 +1,158 @@
+
+	// Fetching the cephCluster
+	cephClusterInstance := &cephv1.CephCluster{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, cephClusterInstance)
+	if errors.IsNotFound(err) {
+		logger.Infof("cephCluster instance not found for %s", request.NamespacedName)
+		return reconcile.Result{}, nil
+	} else if err != nil {
+		logger.Errorf("could not fetch cephCluster %s: %+v", request.Name, err)
+		return reconcile.Result{}, err
+	}
+
+	// skipping the reconcile since the feature is switched off
+	if !cephClusterInstance.Spec.DisruptionManagement.ManageMachineDisruptionBudgets {
+		logger.Debugf("Skipping reconcile for cephCluster %s as manageMachineDisruption is turned off", request.NamespacedName)
+		return reconcile.Result{}, nil
+	}
+
+	mdb := &healthchecking.MachineDisruptionBudget{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      generateMDBInstanceName(request.Name, request.Namespace),
+			Namespace: cephClusterInstance.Spec.DisruptionManagement.MachineDisruptionBudgetNamespace,
+		},
+	}
+
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: mdb.GetName(), Namespace: mdb.GetNamespace()}, mdb)
+	if errors.IsNotFound(err) {
+		// If the MDB is not found creating the MDB for the cephCluster
+		maxUnavailable := int32(0)
+		// Generating the MDB instance for the cephCluster
+		newMDB := &healthchecking.MachineDisruptionBudget{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      generateMDBInstanceName(request.Name, request.Namespace),
+				Namespace: cephClusterInstance.Spec.DisruptionManagement.MachineDisruptionBudgetNamespace,
+				Labels: map[string]string{
+					MDBCephClusterNamespaceLabelKey: request.Namespace,
+					MDBCephClusterNameLabelKey:      request.Name,
+				},
+				OwnerReferences: []metav1.OwnerReference{cephCluster.ClusterOwnerRef(cephClusterInstance.GetName(), string(cephClusterInstance.GetUID()))},
+			},
+			Spec: healthchecking.MachineDisruptionBudgetSpec{
+				MaxUnavailable: &maxUnavailable,
+				Selector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						machinelabel.MachineFencingLabelKey:          request.Name,
+						machinelabel.MachineFencingNamespaceLabelKey: request.Namespace,
+					},
+				},
+			},
+		}
+		err = r.client.Create(context.TODO(), newMDB)
+		if err != nil {
+			logger.Errorf("failed to create mdb %+v", err)
+			return reconcile.Result{}, err
+		}
+		return reconcile.Result{}, nil
+	} else if err != nil {
+		logger.Errorf("%+v", err)
+		return reconcile.Result{}, err
+	}
+	if mdb.Spec.MaxUnavailable == nil {
+		maxUnavailable := int32(0)
+		mdb.Spec.MaxUnavailable = &maxUnavailable
+	}
+	// Check if the cluster is clean or not
+	_, isClean, err := cephClient.IsClusterClean(r.context.ClusterdContext, request.Name)
+	if err != nil {
+		logger.Errorf("failed to get cephCluster status %+v", err)
+		maxUnavailable := int32(0)
+		mdb.Spec.MaxUnavailable = &maxUnavailable
+		updateErr := r.client.Update(context.TODO(), mdb)
```

Comment:
The updateErr is never being logged. Please wrap the reconcile function like this: https://github.com/rook/rook/blob/master/pkg/operator/ceph/disruption/clusterdisruption/reconcile.go#L67